python/lsst/pipe/tasks/zooniverseCutouts.py

```diff
@@ -0,0 +1,224 @@
+
+import lsst.dax.apdb
+import lsst.pex.config as pexConfig
+import lsst.pipe.base
+
+__all__ = ["ZooniverseCutoutsConfig", "ZooniverseCutoutsTask"]
+
+
+class ZooniverseCutoutsConfig(pexConfig.Config):
+    size = pexConfig.Field(
+        doc=("Size of cutout to extract for image from science, template, and difference exposures."
+             " TODO: should we have an option to use the source footprints as the cutout size instead?"),
+        dtype=int,
+        default=30
+    )
+    urlRoot = pexConfig.Field(
+        doc="URL that the resulting images will be served to Zooniverse from.",
+        dtype=str,
+        default=None
+    )
+    asinhScale = pexConfig.Field(
+        doc="Scale factor ('a') in AsinhStretch for template/science images.",
+        dtype=float,
+        default=0.1
+    )
+    # TODO: we'll likely need these for implementing select_sources, but
+    # apdb does not yet have an API for `where` statements.
+    # apdb = pexConfig.ConfigurableField(
+    #     target=lsst.dax.apdb.Apdb,
+    #     ConfigClass=lsst.dax.apdb.ApdbConfig,
+    #     doc="Database connection for retreiving DiaSources."
+    #         "Must already be initialized.",
+    # )
+    # query = pexConfig.Field(
+    #     doc="SQL query to run on APDB to identify sources to extract cutouts of.",
+    #     default="select * from DiaSource LIMIT 100",
+    #     dtype=str
+    # )
+    outputPath = pexConfig.Field(
+        doc=("The full path to write the files to; images will go in `outputPath/images/`, "
+             "while the manifest file will go in `outputPath/`"),
+        default=None,
+        dtype=str
+    )
+
+
+class ZooniverseCutoutsTask(lsst.pipe.base.Task):
+    """Generate cutouts and a manifest for upload to a Zooniverse project.
+    """
+    ConfigClass = ZooniverseCutoutsConfig
+    _DefaultName = "zooniverseCutouts"
+
+    def run(self, butler):
+        """Select sources from apdb and generate zooniverse cutouts an a
+        manifest for upload.
+
+        This method is entirely TODO, and is a placeholder while we sort out
+        select_sources and how we want the butler to be passed in.
+        """
+        data = self.select_sources()
+        self.generate_images(data, butler)
+
+        manifest = self.make_manifest(data)
+        manifest.to_csv(os.path.join(self.config.outputPath, "manifest.csv"), index=False)
+
+    def select_sources(self):
+        """Select sources from apdb to make cutouts of.
+
+        This method is entirely TODO until we have a way to make where queries
+        on apdb.
+        """
+        return None
+
+    def make_manifest(self, data):
+        """Return a Zooniverse manifest attaching image URLs to source ids.
+        """
+        output = pd.DataFrame([])
+        output['external_id'] = data['diaSourceId']
+
+        def make_url(source):
+            """Return a URL for this source."""
+            return f"{self.config.urlRoot.rstrip('/')}/{source['diaSourceId']}.png"
```

Comment:
I wasn't sure this could work, but `DataFrame.apply` can pass other args to the target function. I added a staticmethod and it fixed a bug too (url was missing `images/` subdir)!